### PR TITLE
[ES] 10-2.md Self-Supervised Learning - ClusterFit and PIRL - spanish translation

### PR DESCRIPTION
**Pull request by tiagogiraldo**
Attached is the Spanish translation of the 10-2.md.  It is ready for review.

### CODE REVIEW

File: docs/es/week10/10-2.md [hunk 1]
```diff
@@ -0,0 +1,494 @@
...
+
+<!--However, it is very unclear why performing a non-semantic task should produce good features?. For example, why should we expect to learn about “semantics” while solving something like Jigsaw puzzle? Or why should “predicting hashtags” from images be expected to help in learning a classifier on transfer tasks? Therefore, the question remains. How should we design good pre-training tasks which are well aligned with the transfer tasks?-->
+Sin embargo, no está muy claro por qué la realización de una tarea no semántica debe producir buenas características... Por ejemplo, ¿por qué deberíamos esperar aprender sobre "semántica" mientras resolvemos algo como un rompecabezas? ¿O por qué se espera que "predecir hashtags" a partir de imágenes ayude a aprender un clasificador en tareas de transferencia? Por lo tanto, la pregunta sigue siendo. ¿Cómo deberíamos diseñar buenas tareas de pre-entrenamiento que estén bien alineadas con las tareas de transferencia?
```
Comment (xcastilla) [inline docs/es/week10/10-2.md:20]:
```suggestion
Sin embargo, no está muy claro por qué la realización de una tarea no semántica debe producir buenas características... Por ejemplo, ¿por qué deberíamos esperar aprender sobre "semántica" mientras resolvemos algo como un rompecabezas? ¿O por qué se espera que "predecir hashtags" a partir de imágenes ayude a aprender un clasificador en tareas de transferencia? Por lo tanto, la pregunta sigue siendo: ¿Cómo deberíamos diseñar buenas tareas de pre-entrenamiento que estén bien alineadas con las tareas de transferencia?
```

File: docs/es/week10/10-2.md [hunk 2]
```diff
@@ -0,0 +1,494 @@
...
+
+<!--  *E.g.* exact location of objects, lighting, exact colour-->
+  *P.ej.* la ubicación exacta de los objetos, la iluminación, el color exacto
+
+<!--  * PIRL: Self-supervised learning of Pre-text Invariant Representations-->
+  * ARIP: Aprendizaje auto-supervisado de Representaciones Invariantes de Pretexto
+
+<!--Two ways to achieve the above properties are **Clustering** and **Contrastive Learning**. They have started performing much better than whatever pretext tasks that were designed so far.  One method that belongs to clustering is **ClusterFit** and another falling into invariance is **PIRL**.-->
+
+Dos formas de lograr las propiedades anteriores son **Agrupamiento** y **Aprendizaje Constrastivo**. Que han comenzado a desempeñarse mucho mejor que las tareas de pretexto que se diseñaron hasta ahora. Un método que pertenece a la agrupación en clústeres es **ClusterFit** y otro que cae en invariancia es **ARIP**.
```
Comment (xcastilla) [inline docs/es/week10/10-2.md:60]:
```suggestion
Dos formas de lograr las propiedades anteriores son **Agrupamiento** y **Aprendizaje Constrastivo**, que han comenzado a desempeñarse mucho mejor que las tareas de pretexto que se diseñaron hasta ahora. Un método que pertenece a la agrupación en clústeres es **ClusterFit** y otro que cae en invariancia es **ARIP**.
```
o bien esta otra:

```suggestion
Dos formas de lograr las propiedades anteriores son **Agrupamiento** y **Aprendizaje Constrastivo**. Estos métodos han comenzado a desempeñarse mucho mejor que las tareas de pretexto que se diseñaron hasta ahora. Un método que pertenece a la agrupación en clústeres es **ClusterFit** y otro que cae en invariancia es **ARIP**.
```

File: docs/es/week10/10-2.md [hunk 3]
```diff
@@ -0,0 +1,494 @@
...
...las representaciones de la última capa capturan una propiedad de muy bajo nivel de la señal. Estas capturan cosas como la rotación o algo así. Mientras que lo que se diseña o lo que se espera de estas representaciones es que sean invarintes a estas cosas que debería ser capaz de reconocer un gato, no importa si el gato está en posición vertical o que el gato está, digamos, inclinado como 90 grados. Mientras que cuando estas resolviendo esa tarea de pretexto particular se impone exactamente lo contrario. Estamos diciendo que deberíamos ser capaces de reconocer si esta imagen está en posición vertical o si esta imagen está básicamente girada hacia los lados. Hay muchas excepciones en las que realmente quieres que estas representaciones de bajo nivel sean covariantes y muchas de ellas tienen que ver con las tareas que estás realizando y bastantes tareas en 3D realmente quieren ser predictivas. Así que quieres predecir qué transformaciones de cámara tienes: estás viendo dos vistas del mismo objeto o algo así. Pero a menos que tengas ese tipo de aplicación específica para muchas tareas semánticas, realmente quieres ser invariante a las transformaciones que se usan para usar esa entrada.
```
Comment (xcastilla) [inline docs/es/week10/10-2.md:249]:
```suggestion
Ahora pasamos a ARPI un poco, y eso es tratar de entender cuál es la principal diferencia de las tareas de pretexto y cómo el aprendizaje contrastivo es muy diferente de las tareas de pretexto. De nuevo, las tareas con pretexto siempre razonan sobre una sola imagen a la vez. Así que la idea es que dándole a una imagen su y previa transformación a esa imagen, en este caso una transformación de rompecabezas, y luego introduciendo esta imagen transformada en una ConvNet e intentando predecir la propiedad de la transformación que aplicaste, la permutación que aplicaste o la rotación que aplicaste o el tipo de color que removiste y así sucesivamente. Así que las tareas de pretexto siempre razonan sobre una sola imagen. Y la segunda cosa es que la tarea que estás realizando en este caso realmente tiene que capturar alguna propiedad de la transformación. Así que realmente tiene que capturar la permutación exacta que se aplican o el tipo de rotación que se aplican, lo que significa que las representaciones de la última capa van a ir mucho ARPI como la transformación de los cambios y eso es por diseño, porque realmente estás tratando de resolver las tareas de pretexto. Pero desafortunadamente, lo que esto significa es que las representaciones de la última capa capturan una propiedad de muy bajo nivel de la señal. Estas capturan cosas como la rotación o algo así. Mientras que lo que se diseña o lo que se espera de estas representaciones es que sean invarintes a estas cosas que debería ser capaz de reconocer un gato, no importa si el gato está en posición vertical o que el gato esté, digamos, inclinado como 90 grados. Mientras que cuando estás resolviendo esa tarea de pretexto particular se impone exactamente lo contrario. Estamos diciendo que deberíamos ser capaces de reconocer si esta imagen está en posición vertical o si esta imagen está básicamente girada hacia los lados. Hay muchas excepciones en las que realmente quieres que estas representaciones de bajo nivel sean covariantes y muchas de ellas tienen que ver con las tareas que estás realizando y bastantes tareas en 3D realmente quieren ser predictivas. Así que quieres predecir qué transformaciones de cámara tienes: estás viendo dos vistas del mismo objeto o algo así. Pero a menos que tengas ese tipo de aplicación específica para muchas tareas semánticas, realmente quieres ser invariante a las transformaciones que se utilizan para usar esa entrada.
```

File: docs/es/week10/10-2.md [hunk 4]
```diff
@@ -0,0 +1,494 @@
...
+height="70%" width="70%"/><br><b>Fig. 20</b>: <!--Pre-training on uncurated YFCC images-->Preentrenamiento en imágenes no curadas del YFCC
+</center>
+
+
+<!--### Semantic Features-->
+### Características semánticas
+
+<!--Now, going back to verifying the semantic features, we look at the Top-1 accuracy for PIRL and Jigsaw for different layers of representation from `conv1` to `res5`. It's interesting to note that the accuracy keeps increasing for different layers for both PIRL and Jigsaw, but drops in the 5th layer for Jigsaw. Whereas, the accuracy keeps improving for PIRL, *i.e.* more and more semantic.-->
+
+Ahora, volviendo a la verificación de los características semánticas, miramos la precisión Top-1 para el ARPI y el rompecabezas para diferentes capas de representación desde "conv1" a "res5". Es interesante observar que la precisión sigue aumentando para las diferentes capas tanto para el ARPI como para el rompecabezas, pero disminuye en la 5ª capa para el rompecabezas. Mientras que, la precisión sigue mejorando para PIRL, *i.e.*, más y más semántica.
```
Comment (xcastilla) [inline docs/es/week10/10-2.md:365]:
```suggestion
Ahora, volviendo a la verificación de los características semánticas, miramos la precisión Top-1 para el ARPI y el rompecabezas para diferentes capas de representación desde "conv1" a "res5". Es interesante observar que la precisión sigue aumentando para las diferentes capas tanto para el ARPI como para el rompecabezas, pero disminuye en la 5ª capa para el rompecabezas. Mientras que, la precisión sigue mejorando para ARPI, *i.e.*, más y más semántica.
```

File: docs/es/week10/10-2.md [hunk 5]
```diff
@@ -0,0 +1,494 @@
...
+
+<center>
+<img src="{{site.baseurl}}/images/week10/10-2/fig21.png"
+height="70%" width="70%"/><br><b>Fig. 21</b>: <!--Quality of PIRL representations per layer-->Calidad de las representaciones del ARPI por capa
+</center>
+
+
+### Scalability
+
+PIRL is very good at handling problem complexity because you're never predicting the number of permutations, you're just using them as input. So, PIRL can easily scale to all 362,880 possible permutations in the 9 patches. Whereas in Jigsaw, since you're predicting that, you're limited by the size of your output space.
```
Comment (xcastilla) [inline docs/es/week10/10-2.md:375]: It seems you skipped the translation of this part 😄

File: docs/es/week10/10-2.md [hunk 6]
```diff
@@ -0,0 +1,494 @@
...
+<!--### Shortcomings-->
+### Limitaciones
+
+<!--1. It's not very clear as to which set of data transforms matter. Although Jigsaw works, it's not very clear why it works.
+2. Saturation with model size and data size.
+3. What invariances matter? (One could think about what invariances work for a particular supervised task in general as future work.)-->
+
+1. No está muy claro qué conjunto para transformación de datos importa.  Aunque el Rompecabezas funciona, no está muy claro por qué funciona.
+2. Saturación con el tamaño del modelo y el tamaño de los datos.
+3. ¿Qué invariancias importa? (Uno podría pensar en qué invariancias funcionan para una tarea supervisada en general como trabajo futuro).
```
Comment (xcastilla) [inline docs/es/week10/10-2.md:407]:
```suggestion
3. ¿Qué invariancias importan? (Uno podría pensar en qué invariancias funcionan para una tarea supervisada en general como trabajo futuro).
```

File: docs/es/week10/10-2.md [hunk 7]
```diff
@@ -0,0 +1,494 @@
...
+<!--### Contrastive learning and batch norms-->
+### Aprendizaje contrastivo y normas de lote
+
+<!--1. Wouldn't the network learn only a very trivial way of separating the negatives from the positives if the contrasting network uses the batch norm layer (as the information would then pass from one sample to the other)?-->
+
+1. ¿No aprendería la red sólo una forma muy trivial de separar los negativos de los positivos si la red de contraste utiliza la capa de norma de lote (ya que la información pasaría entonces de una muestra a la otra)?
+
+<!--**Ans**: *In PIRL, no such phenomenon was observed, so just the usual batch norm was used*-->
+
+**Resp**: *En e ARPI, no se observó tal fenómeno, así que sólo se utilizó la norma habitual de lotes*
```
Comment (xcastilla) [inline docs/es/week10/10-2.md:425]:
```suggestion
**Resp**: *En ARPI, no se observó tal fenómeno, así que sólo se utilizó la norma habitual de lotes*
```

File: docs/es/week10/10-2.md [hunk 8]
```diff
@@ -0,0 +1,494 @@
...
+
+2. Entonces, ¿está bien usar normas de lote para cualquier red de contraste?
+
+<!--**Ans**: *In general, yeah. In SimCLR, a variant of the usual batch norm is used to emulate a large batch size. So, batch norm with maybe some tweaking could be used to make the training easier*-->
+
+**Resp**: *En general, sí. En SimCLR, se utiliza una variante de la norma de lote habitual para emular un gran tamaño de lote. Por lo tanto, la norma de lote con tal vez algunos ajustes podría ser utilizado para hacer el entrenamiento más sencillo *
+
+<!--3. Does the batch norm work in the PIRL paper only because it's implemented as a memory bank - as all the representations aren't taken at the same time? (As batch norms aren't specifically used in the MoCo paper for instance)-->
+
+3. ¿Funciona la norma de lote en el papel en el ARPI sólo porque se implementa como un banco de memoria - ya que todas las representaciones no se toman al mismo tiempo? (Como las normas de lote no se utilizan específicamente en el papel MoCo, por ejemplo)
```
Comment (xcastilla) [inline docs/es/week10/10-2.md:437]:
```suggestion
3. ¿Funciona la norma de lote en el papel en el ARPI sólo porque se implementa como un banco de memoria - ya que todas las representaciones no se toman al mismo tiempo? (Como las normas de lote no se utilizan específicamente en el artículo de MoCo, por ejemplo)
```

File: docs/es/week10/10-2.md [hunk 9]
```diff
@@ -0,0 +1,494 @@
...
+
+<!--**Ans**: *Yeah. In PIRL, the same batch doesn't have all the representations and possibly why batch norm works here, which might not be the case for other tasks where the representations are all correlated within the batch*-->
+
+**Resp**: *Sí. En el PIRL, el mismo lote no tiene todas las representaciones y posiblemente por qué la norma del lote funciona aquí, lo que podría no ser el caso para otras tareas donde las representaciones están todas correlacionadas dentro del lote*
```
Comment (xcastilla) [inline docs/es/week10/10-2.md:441]:
```suggestion
**Resp**: *Sí. En el ARPI, el mismo lote no tiene todas las representaciones y posiblemente por qué la norma del lote funciona aquí, lo que podría no ser el caso para otras tareas donde las representaciones están todas correlacionadas dentro del lote*
```

**Comment by xcastilla** [comment]
You will need to add the path to this lesson (10-2) in [this file](https://github.com/Atcold/pytorch-Deep-Learning/blob/b8c8188d9d3774b84368f31c53412a0713435013/docs/_config.yml) as well

**Review by xcastilla — CHANGES_REQUESTED** [review]
I left some comments. Some of the suggestions are very trivial but I thought might improve readability. Nice translation anyway, it was really easy to read 👍

**Review by tiagogiraldo — COMMENTED** [review]
The changes are ready, except for the missing translation fragment that I upload in a moment.

**Comment by JonathanSum** [comment]
Don't forget to put [ES] at the starting of the title.

**Review by xcastilla — APPROVED** [review]
Looks good!

File: docs/es/week10/10-2.md [hunk 10]
```diff
@@ -0,0 +1,499 @@
+---
+lang-ref: ch.10-2
+title: Self-Supervised Learning - ClusterFit and PIRL
```
Comment (Atcold) [inline docs/es/week10/10-2.md:3]: Title needs translation.

File: docs/es/week10/10-2.md [hunk 11]
```diff
@@ -0,0 +1,499 @@
...
+lecturer: Ishan Misra
+authors: Zhonghui Hu, Yuqing Wang, Alfred Ajay Aureate Rajakumar, Param Shah
+date: 6 Apr 2020
+lang: es
+translation-date:
```
Comment (Atcold) [inline docs/es/week10/10-2.md:8]: Missing date.

**Review by Atcold — CHANGES_REQUESTED** [review]
You need to update `docs/_config.yml` as well.

File: docs/es/week10/10-2.md (hunk 11 above)
Comment (tiagogiraldo) [reply docs/es/week10/10-2.md]: I've already made the change

File: docs/es/week10/10-2.md (hunk 10 above)
Comment (tiagogiraldo) [reply docs/es/week10/10-2.md]: I've already made the change

**Review by Atcold — APPROVED** [review]
Thanks! I've updated `docs/_config.yml` for you.

**Pull request merged**